wp-content/mu-plugins/pantheon/pantheon-updates.php

```diff
@@ -3,39 +3,64 @@
 if( ! defined('WP_AUTO_UPDATE_CORE')) {
 	define( 'WP_AUTO_UPDATE_CORE', false );
 }
-remove_action( 'wp_maybe_auto_update', 'wp_maybe_auto_update' );
 
-// Remove the default WordPress core update nag if on Pantheon
-if( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) ){
-    add_action('admin_menu','_pantheon_hide_update_nag');
+// Short circuit WordPress checking WordPress.org for updates by returning zero updates and current time as last checked time
+function _pantheon_disable_wp_updates() {
+	// include an unmodified $wp_version
+	include( ABSPATH . WPINC . '/version.php' );
+
+	return (object) array(
+		'updates' => array(),
+		'version_checked' => $wp_version,
+		'last_checked' => time(),
+	);
+
 }
 
 function _pantheon_hide_update_nag() {
 	remove_action( 'admin_notices', 'update_nag', 3 );
+	remove_action( 'network_admin_notices', 'update_nag', 3 );
+}
+
+// Remove the default WordPress core update nag if on Pantheon
+if( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) ){
+	add_filter( 'pre_site_transient_update_core', '_pantheon_disable_wp_updates' );
+    add_action('admin_menu','_pantheon_hide_update_nag');
 }
 
-// Get the latest WordPress version
+// Get the latest WordPress version tagged in the Pantheon upstream on GitHub
 function _pantheon_get_latest_wordpress_version() {
-	$core_updates = get_core_updates( array('dismissed' => false) );
+	$pantheon_latest_wp_version = get_transient( 'pantheon_latest_wp_version' );
+
+	if( false === $pantheon_latest_wp_version ){
+		$github_api_pantheon_upstream_tags_url = "https://api.github.com/repos/pantheon-systems/WordPress/git/refs/tags";
```

Comment:
Perhaps we should make this filterable for folks using a custom upstream.

Reply:
Is the custom upstream exposed by the Pantheon platform to the WordPress site? It would be better to handle this automatically for the user rather than make everyone with a custom upstream write a hook.

If we can't tell what the custom upstream is, I'd suggest that we solve this only for Pantheon's upstream for now, and leave custom upstreams as follow-on work.

Reply:
I'm not sure if we can tell what the upstream is. I don't see it in [environment variables](https://pantheon.io/docs/read-environment-config/) and am not sure how much work it would be to add.

My concern here is if folks fork our upstream for a custom upstream they will get this `mu-plugin` and the update nag. If we can't detect the upstream we won't be able to automatically opt out for custom upstreams.

A WordPress filter would at least let them update manually without hacking the file. We could also create a filter to fire the nag at all, defaulting to `true` but allowing an override. This would be useful for sites where WordPress is installed with Composer, pushed from CI or other means besides merging the Pantheon upstream directly on the dashboard.

Reply:
Today's behavior is that users with a custom upstream will get an update nag when WordPress core has an update available, but Pantheon does not yet have a release available and therefore there is no update available for their custom upstream. In this case, they can do nothing.

With this PR, folks will not get the nag until Pantheon has a release available. If there is not yet a release available for the custom upstream, at least the user can now go and update the upstream.

If we did as you suggest and remove the nag until the custom upstream has an update available, then perhaps the utility of the nag would be eliminated, as the user might not know that they need to update the upstream.

I also see the utility in what you suggest. I think we should move forward just making this one step better (tying the nag with the availability of Pantheon's update), and then evaluate the use-case for improving the nag and how to implement it as a follow-on task. I think that most agencies are not going to want to patch their WordPress sites just to optimize the nag message; they're going to want us to do that for them. But let's discuss that separately.